.circleci/config.yml

```diff
@@ -0,0 +1,139 @@
+      run: bundle install --jobs 4 --path=vendor/bundle --retry 3
+    - run: bundle exec appraisal install
+    - &cache_gems
+      save_cache:
+        key: v2-gems-{{ checksum "Gemfile" }}-{{ checksum "Appraisals"}}-{{ .Environment.CIRCLE_JOB }}
+        paths:
+          - ~/loga/gemfiles/vendor/bundle
+          - ~/loga/vendor/bundle
+    - attach_workspace:
+        at: ./tmp
+    - run: RACK_ENV=development bundle exec appraisal rspec
+    - run: RACK_ENV=production bundle exec appraisal rspec
+    - run: ./tmp/cc-test-reporter format-coverage -t simplecov -o $(ruby -e 'puts "tmp/codeclimate.#{RUBY_VERSION}.json"') coverage/.resultset.json
+    - persist_to_workspace:
+        root: tmp
+        paths:
+          - codeclimate.*.json
+    - store_artifacts:
+        path: coverage
+
+version: 2
+jobs:
+  build:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - run:
+          name: Download cc-test-reporter
+          command: |
+            mkdir -p tmp/
+            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./tmp/cc-test-reporter
+            chmod +x ./tmp/cc-test-reporter
+      - persist_to_workspace:
+          root: ./tmp
+          paths:
+            - cc-test-reporter
+  upload-coverage:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - attach_workspace:
+          at: ./tmp
+      - run:
+          name: Upload coverage results to Code Climate
+          command: |
+            ./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json -o tmp/codeclimate.total.json
+            ./tmp/cc-test-reporter upload-coverage -i tmp/codeclimate.total.json -r $CODECLIMATE_REPO_TOKEN
+  ruby-2.2:
+    docker:
+      - image: circleci/ruby:2.2.10
+    <<: *defaults
+  ruby-2.3:
+    docker:
+      - image: circleci/ruby:2.3.7
+    <<: *defaults
+  ruby-2.4:
+    docker:
+      - image: circleci/ruby:2.4.4
+    <<: *defaults
+  ruby-2.5:
+    docker:
+      - image: circleci/ruby:2.5.1
+    <<: *defaults
+  rubocop:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - checkout
+      - *restore_gems
+      - *bundle_install
+      - *cache_gems
+      - run: bundle exec rubocop
+  push-to-rubygems:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - checkout
+      - *restore_gems
+      - *bundle_install
+      - *cache_gems
+      - run:
+          name: Build gem
+          command: gem build $CIRCLE_PROJECT_REPONAME.gemspec
+      - run:
+          name: Publish to rubygems
+          command: |
+            echo ":rubygems_api_key: $RUBYGEMS_API_KEY" >  ~/.gem/credentials
+            chmod 0600 ~/.gem/credentials
+            gem push $CIRCLE_PROJECT_REPONAME-$(echo $CIRCLE_TAG | sed -e 's/v//').gem
+workflows:
+  version: 2
+  test-then-push-gem:
+    jobs:
+      - build
```

Comment:
🙅 There's a important thing to remember with CircleCI 2.0 workflows:

> CircleCI treats tag and branch filters differently when deciding whether a job should run.
>
> 1. For a branch push unaffected by any filters, CircleCI runs the job.
> 1. For a tag push unaffected by any filters, CircleCI skips the job.
>
> Item two above means that a job **must** have a `filters tags` section to run as a part of a tag push and all its transitively dependent jobs **must** also have a `filters tags` section.

https://circleci.com/docs/2.0/workflows/#git-tag-job-execution

Reply:
So in order for the `push-to-rubygems` job to trigger, the `build` job needs to be configured something like:
```yaml
- build:
    filters:
      tags:
        only: /.+/
```